forge-ai/src/main/java/forge/ai/ability/AttachAi.java

```diff
@@ -833,7 +833,27 @@ private static Card attachAICursePreference(final SpellAbility sa, final List<Ca
         int totPower = 0;
         final List<String> keywords = new ArrayList<>();
 
+        boolean cantAttack = false;
+        boolean cantBlock = false;
+
         for (final StaticAbility stAbility : attachSource.getStaticAbilities()) {
+            if (stAbility.getParam("Mode").equals(StaticAbilityCantAttackBlock.CantAttackMode)) {
+                String valid = stAbility.getParam("ValidCard");
+                if (valid.contains(stCheck) || valid.contains("AttachedBy")) {
```

Comment:
I wonder if we should just remove Enchanted vs. EquippedBy
technically it shouldn't matter for these cases:
> 301.5f. An ability of a permanent that refers to the "equipped creature" refers to whatever creature that permanent is attached to, even if the permanent with the ability isn't an Equipment.

Reply:
maybe yes, but i still think it helps the AI for now

we can clean that up later

Reply:
@tool4ever what we could do is to cleanup the Card Properties there, and their negations:

https://github.com/Card-Forge/forge/blob/194662c9c7155a52513936bd9b277087b621440b/forge-game/src/main/java/forge/game/card/CardProperty.java#L1373-L1396